src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -124,11 +173,20 @@ public string GetConfigFileName()
     /// <param name="replaceEnvVar">Whether to replace environment variable with its
     /// value or not while deserializing.</param>
     /// <returns>True if the config was loaded, otherwise false.</returns>
-    public override bool TryLoadKnownConfig([NotNullWhen(true)] out RuntimeConfig? config, bool replaceEnvVar = false)
+    public override bool TryLoadKnownConfig([NotNullWhen(true)] out RuntimeConfig? config, bool replaceEnvVar = false, string? defaultDataSourceName = null)
     {
         return TryLoadConfig(ConfigFilePath, out config, replaceEnvVar);
     }
 
+    /// <summary>
+    /// Hot Reloads the runtime config when the file watcher
+    /// is active and detects a change to the underlying config file.
+    /// </summary>
+    public void HotReloadConfig(string defaultDataSourceName)
```

Comment:
why is the param named `defaultDataSourceName`? what happens when the value is not from the default data source and, instead, from a different data source?